qiskit_experiments/framework/experiment_data.py

```diff
@@ -69,23 +88,129 @@ def completion_times(self) -> Dict[str, datetime]:
 
         return job_times
 
-    @classmethod
-    def load(cls, experiment_id: str, service: DatabaseServiceV1) -> "ExperimentData":
-        """Load a saved experiment data from a database service.
+    def add_child_data(self, experiment_data: ExperimentData):
+        """Add child experiment data to the current experiment data"""
+        experiment_data._parent_id = self.experiment_id
+        self._child_data[experiment_data.experiment_id] = experiment_data
+
+    def child_data(
+        self, index: Optional[Union[int, slice, str]] = None
+    ) -> Union[ExperimentData, List[ExperimentData]]:
+        """Return child experiment data.
 
         Args:
-            experiment_id: Experiment ID.
-            service: the database service.
+            index: Index of the child experiment data to be returned.
+                Several types are accepted for convenience:
+
+                    * None: Return all child data.
+                    * int: Specific index of the child data.
+                    * slice: A list slice of indexes.
+                    * str: experiment ID of the child data.
 
         Returns:
-            The loaded experiment data.
+            The requested single or list of child experiment data.
+
+        Raises:
+            QiskitError: if the index or ID of the child experiment data
+                         cannot be found.
         """
-        expdata = DbExperimentDataV1.load(experiment_id, service)
+        if index is None:
+            return self._child_data.values()
+        if isinstance(index, (int, slice)):
+            return self._child_data.values()[index]
+        if isinstance(index, str):
+            return self._child_data[index]
+        raise QiskitError(f"Invalid index type {type(index)}.")
+
+    def component_experiment_data(
+        self, index: Optional[Union[int, slice]] = None
+    ) -> Union[ExperimentData, List[ExperimentData]]:
+        """Return child experiment data"""
+        warnings.warn(
+            "This method is deprecated and will be removed next release. "
+            "Use the `child_data` method instead.",
+            DeprecationWarning,
+        )
+        return self.child_data(index)
+
+    def save(self) -> None:
+        super().save()
+        for data in self._child_data.values():
+            original_verbose = data.verbose
+            data.verbose = False
+            data.save()
+            data.verbose = original_verbose
+
+    def save_metadata(self) -> None:
+        # Copy child experiment IDs to metadata
+        if self._child_data:
```

Comment:
The intent was these IDs are only needed for saving and retrieving from DB so are localized to the save and load functions. The complication with putting it else where is the child IDs can change anytime analysis is re-run, so you would have to track this which I think would make things more complicated. This is why the initialize_components stores the indices of component child experiments, rather than the IDs, so it can look up the current ID values each time analysis is re-run.